File: FsUniversityApi/Database/Entities/Person.fs
namespace FsUniversityApi.Database.Entities

open System.ComponentModel.DataAnnotations
open System
open FsUniversityApi.Database.Entities.ISoftRemovableEntity
open ApiDtoLibrary.Person

module Person =
    [<CLIMutable>]
    type PersonInfo =
        {
            [<Key>]
            PersonInfoId : Guid

            [<Required>]
            ApplicationUserId : Guid

            [<Required>]
            FirstName : string

            [<Required>]
            LastName : string

            [<Required>]
            PESEL: string

            Birthday : DateTime
            MotherLand : string
            PersonStatus : PersonStatus
            SoftDeleted : bool
        }
        interface ISoftRemovableEntity with
            member x.SoftDeleted = x.SoftDeleted

    type IPerson = 
        abstract PersonInfo : PersonInfo

File: GrpcService/Models/Person.cs
using ApiDtoLibrary.Person;

namespace GrpcService.Models
{
    public class Person
    {
        public Guid PersonId { get; set; }
        public Guid ApplicationUserId { get; set; } = Guid.Empty;

        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string? PESEL { get; set; }
        public DateTime? Birthday { get; set; }
        public string? Motherland { get; set; }

        public PersonStatus PersonStatus { get; set; } = PersonStatus.Unidentified;

        public bool SoftDeleted { get; set; } = false;
    }
}
